app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -135,8 +135,20 @@ def on_shutdown
         @task.run_callbacks(:interrupt)
       end
 
+      db_lock_version = @run.class.where(id: @run.id).pluck(:lock_version).first
+      if db_lock_version != @run.lock_version
+        raise ActiveRecord::StaleObjectError, @run
+      end
```

Comment:
I don't think we should read the lock version, because we're always going to end up with a potential race condition between that read and the write in `after_perform`.